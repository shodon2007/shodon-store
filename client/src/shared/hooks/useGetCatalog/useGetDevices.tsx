import { useQuery } from "@tanstack/react-query"
import productApi from "src/shared/api/productApi";

const useGetDevices = (type: string) => {
    const query = useQuery({
        queryKey: ["products", type],
        queryFn: async () => {
            const res = await productApi.getAllProducts(type);
            return res.data;
        }
    });

    return query;
}

export {
    useGetDevices
}