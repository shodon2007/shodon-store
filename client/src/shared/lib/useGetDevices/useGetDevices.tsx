import { useQuery } from '@tanstack/react-query';
import { FilterType } from 'src/app/types/filter';
import productApi from 'src/shared/api/productApi';

const useGetDevices = (filters: FilterType, type: string) => {
	const query = useQuery({
		queryKey: ['products', type],
		queryFn: async () => {
			const res = await productApi.getAllProducts(filters, type);
			return res.data;
		},
		retry: 1,
	});

	return query;
};

export { useGetDevices };
