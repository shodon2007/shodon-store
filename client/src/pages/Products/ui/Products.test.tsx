import { screen } from '@testing-library/react';
import * as useGetDevicesModule from 'src/shared/lib/useGetDevices/useGetDevices';
import renderTestApp from 'src/app/tests/helpers/renderTestApp';

describe('testing Devices.tsx', () => {
	const devices: any = {
		data: [
			{
				name_ru: 'hello baby ',
				img: 'dsfdsfsfsf',
				id: 1,
				name: 'shodon store',
				price: 150000,
				brand_id: 1,
				rate: '5',
				reviews: 'dslkfjdsklfj',
				type_id: 15,
				attributes: [
					{
						id: 1,
						title: 'ОЗУ',
						description: '15gb',
					},
				],
			},
			{
				name_ru: 'hello baby ',
				img: 'dsfdsfsfsf',
				id: 2,
				name: 'superdfjklsdjf',
				price: 150000,
				brand_id: 1,
				rate: '5',
				reviews: 'dslkfjdsklfj',
				type_id: 15,
				attributes: [
					{
						id: 1,
						title: 'ОЗУ',
						description: '15gb',
					},
					{
						id: 2,
						title: 'память',
						description: '12gb',
					},
				],
			},
		],
		isLoading: false,
		isError: false,
		refetch: Function,
	};

	test('testing Device List', async () => {
		jest.spyOn(useGetDevicesModule, 'useGetDevices').mockReturnValue(devices);
		renderTestApp('/phones');
		const devicesPage = await screen.findByTestId('devices-page');
		const deviceItem = await screen.findAllByTestId('device-item');
		expect(devicesPage).toBeInTheDocument();
		expect(devicesPage).toMatchSnapshot();
		expect(deviceItem.length).toBe(2);
	});

	test('testing spinner', async () => {
		const mockDevices: any = {
			isFetching: true,
			refetch: (): any => null,
		};
		jest
			.spyOn(useGetDevicesModule, 'useGetDevices')
			.mockReturnValue(mockDevices);
		renderTestApp('/phones');

		expect(screen.getByTestId('spinner')).toBeInTheDocument();
	});
});
