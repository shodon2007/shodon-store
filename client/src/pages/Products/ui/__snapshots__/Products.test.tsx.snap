// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing Devices.tsx testing Device List 1`] = `
<div
  class="devicesPage"
  data-testid="devices-page"
>
  <div>
    Загрузка фильтров...
  </div>
  <div
    class="devices"
  >
    <div
      class="device"
      data-testid="device-item"
    >
      <div
        class="left"
      >
        <img
          class="image"
          src="https://store-server.shodon.ru/imgdsfdsfsfsf"
        />
      </div>
      <div
        class="center"
      >
        <div
          class="top"
        >
          <span
            class="Title "
            data-testid="title"
          >
            shodon store
          </span>
          <div
            class="attributes"
          >
            <div
              class="attribute"
            >
              <span
                class="attribute_title"
              >
                ОЗУ
              </span>
              :
              <span
                class="attribute_description"
              >
                15gb
              </span>
            </div>
          </div>
        </div>
        <div
          class="bottom"
        >
          <div>
            NaN
             отзывов
          </div>
          <div
            class="rate"
          >
            <div
              class="container"
            >
              <div
                class="stars"
                style="width: 100%;"
              >
                <svgrurl
                  class="star"
                />
                <svgrurl
                  class="star"
                />
                <svgrurl
                  class="star"
                />
                <svgrurl
                  class="star"
                />
                <svgrurl
                  class="star"
                />
              </div>
            </div>
            <svgrurl
              class="star"
            />
            <svgrurl
              class="star"
            />
            <svgrurl
              class="star"
            />
            <svgrurl
              class="star"
            />
            <svgrurl
              class="star"
            />
          </div>
        </div>
      </div>
      <div
        class="right"
      >
        <span
          class="Title "
          data-testid="title"
        >
          150000
          ₽
        </span>
        <button
          class="Button "
        >
          В корзину
        </button>
      </div>
    </div>
    <div
      class="device"
      data-testid="device-item"
    >
      <div
        class="left"
      >
        <img
          class="image"
          src="https://store-server.shodon.ru/imgdsfdsfsfsf"
        />
      </div>
      <div
        class="center"
      >
        <div
          class="top"
        >
          <span
            class="Title "
            data-testid="title"
          >
            superdfjklsdjf
          </span>
          <div
            class="attributes"
          >
            <div
              class="attribute"
            >
              <span
                class="attribute_title"
              >
                ОЗУ
              </span>
              :
              <span
                class="attribute_description"
              >
                15gb
              </span>
            </div>
            <div
              class="attribute"
            >
              <span
                class="attribute_title"
              >
                память
              </span>
              :
              <span
                class="attribute_description"
              >
                12gb
              </span>
            </div>
          </div>
        </div>
        <div
          class="bottom"
        >
          <div>
            NaN
             отзывов
          </div>
          <div
            class="rate"
          >
            <div
              class="container"
            >
              <div
                class="stars"
                style="width: 100%;"
              >
                <svgrurl
                  class="star"
                />
                <svgrurl
                  class="star"
                />
                <svgrurl
                  class="star"
                />
                <svgrurl
                  class="star"
                />
                <svgrurl
                  class="star"
                />
              </div>
            </div>
            <svgrurl
              class="star"
            />
            <svgrurl
              class="star"
            />
            <svgrurl
              class="star"
            />
            <svgrurl
              class="star"
            />
            <svgrurl
              class="star"
            />
          </div>
        </div>
      </div>
      <div
        class="right"
      >
        <span
          class="Title "
          data-testid="title"
        >
          150000
          ₽
        </span>
        <button
          class="Button "
        >
          В корзину
        </button>
      </div>
    </div>
  </div>
</div>
`;
